src/xpcc/architecture/platform/board/nucleo_l476rg/nucleo_l476rg.hpp

```diff
@@ -0,0 +1,99 @@
+// coding: utf-8
+/* Copyright (c) 2017, Sascha Schade (strongly-typed)
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ */
+// ----------------------------------------------------------------------------
+
+//
+// NUCLEO-L476RG
+// Nucleo kit for STM32L476RG
+// www.st.com/web/catalog/tools/FM146/LN1847/PF261636
+//
+
+#ifndef XPCC_STM32_NUCLEO_L476RG_HPP
+#define XPCC_STM32_NUCLEO_L476RG_HPP
+
+#include <xpcc/architecture/platform.hpp>
+#include <xpcc/debug/logger.hpp>
+
+using namespace xpcc::stm32;
+
+namespace Board
+{
+
+/// STM32L4 running at 48MHz generated from the
+/// internal Multispeed oscillator
+
+// Dummy clock for devices
+struct systemClock {
+	static constexpr uint32_t Frequency = 48 * MHz1;
+	static constexpr uint32_t Ahb  = Frequency;
+	static constexpr uint32_t Apb1 = Frequency;
+	static constexpr uint32_t Apb2 = Frequency;
+
+	static constexpr uint32_t Spi1 = Apb2;
+
+	static constexpr uint32_t Spi2 = Apb1;
+	static constexpr uint32_t Spi3 = Apb1;
+
+	static constexpr uint32_t Can1 = Apb1;
+
+	static constexpr uint32_t Usart1 = Apb2;
+
+	static constexpr uint32_t Usart2 = Apb1;
+	static constexpr uint32_t Usart3 = Apb1;
+	static constexpr uint32_t Usart4 = Apb1;
+	static constexpr uint32_t Usart5 = Apb1;
+
+	static bool inline
+	enable()
+	{
+		// set flash latency first because system already runs from MSI
+		ClockControl::setFlashLatency(Frequency);
+
+		ClockControl::enableMultiSpeedInternalClock(ClockControl::MsiFrequency::MHz48);
+
+		xpcc::clock::fcpu     = Frequency;
+		xpcc::clock::fcpu_kHz = Frequency / 1000;
+		xpcc::clock::fcpu_MHz = Frequency / 1000000;
+		xpcc::clock::ns_per_loop = 47; // 3000 / 64 = 31.125 = ~31ns per delay loop
```

Comment:
`clock.cpp.in` initializes this value for the boot-up frequency, so that delays work before the clock is initialized (could be never).
If you want a more implicit way: `::round(3000 / (Frequency / 1000000));` Though I'm not certain of the `3000` for L4, that depends on the CPU config (pipeline, branching, cache etc), though I assume the L4 is just a CM4 with low-power silicon process.